src/test/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/DynamicProgramming/DPBasedOptimizerHelperTest.java

```diff
@@ -0,0 +1,58 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.DynamicProgramming;
+
+import org.junit.Test;
+import se.liu.ida.hefquin.engine.EngineTestBase;
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.utils.PhysicalPlanFactory;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.simple.DPBasedJoinPlanOptimizer;
+import se.liu.ida.hefquin.engine.utils.Pair;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class DPBasedOptimizerHelperTest extends EngineTestBase {
+    final List<Integer> list= List.of( new Integer[]{1, 2, 3, 4, 5} );
+
+    @Test
+    public void getSubsets_1()
+    {
+        final List<List<Integer>> subsets = DPBasedJoinPlanOptimizer.getSubSet( list, 3 );
+
+        assertEquals( 10, subsets.size() );
+        subsets.forEach( System.out::println );
```

Comment:
Unit tests should better not have such side effects (printing to std.out). Instead, for the given case, it would be better if the unit test actually checks (automatically) that `subsets` contains all the expected subsets.